### ISSUE TITLE
Input data coding and QCFS derivative 

### ISSUE THREAD

**Issue by MusheerAbdullah** [putshua/SNN-conversion-QCFS#12]
Hello Sir, Hope you are alright 
I am just wondering about your code and results that is shown in your paper, 
Q1: Do you use encoding mechanism to convert the input images into spikes during SNN evaluation, or you just apply the original images to the first layer T times ? 

Q2: In your paper (equation 17 ),  you mentioned the derivative functions for training your QCFS, but I cannot find that in your code, and you apply the autograde, am I right ? if yes, is your mentioned results based on equation 17 or using the autograde derivations ? 

waiting for your reply, 
Thank you !

**Comment by putshua** [comment]
Hello,

A1: Yes, I directly applied the original images to the first layer.

A2: I use pytorch autograd. Also, the derivative functions in the original paper in openreview is incorrect. I have corrected them in a new arxiv version.

Thanks for your attention.

Tong Bu


发件人: MusheerAbdullah
发送时间: 2023年8月24日 12:04
收件人: putshua/SNN_conversion_QCFS
抄送: Subscribed
主题: [putshua/SNN_conversion_QCFS] Input data coding and QCFS derivative (Issue #12)

Hello Sir, Hope you are alright
I am just wondering in your code and results shown in your paper,
Q1: Do you use encoding mechanism to convert the input images into spikes during SNN evaluation, or you just apply the original images to the first layer T times.
Q2: in your paper (equation 17 ) you mentioned the derivative functions for training your QCFS, but I cannot find that in your code, and you apply the autograde, am I right ? and is your mentioned results based on equation 17 or using the autograde derivations
waiting for your reply,
Thank you !

**Comment by MusheerAbdullah** [comment]
Thank you so much for your response, I appreciate it.

Can you please share the link of your updated version with the correct derivative? 


Thank you again

**Comment by putshua** [comment]
[[2303.04347] Optimal ANN-SNN Conversion for High-accuracy and Ultra-low-latency Spiking Neural Networks (arxiv.org)](https://arxiv.org/abs/2303.04347)

**Comment by MusheerAbdullah** [comment]
Thank you so much
